pkg/util/trace/trace.go

```diff
@@ -0,0 +1,49 @@
+package trace
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/utils/trace"
+)
+
+type Tracer struct {
+	traceMap map[string]*trace.Trace
+	mux      sync.Mutex
+
+	Threshold time.Duration
+}
+
+func (t *Tracer) StartTrace(key string, name string, field ...trace.Field) {
+	t.mux.Lock()
+	defer t.mux.Unlock()
+	if t.traceMap == nil {
+		t.traceMap = make(map[string]*trace.Trace)
+	}
+	t.traceMap[key] = trace.New(name, field...)
+	return
+}
+
+func (t *Tracer) StopTrace(key string) {
+	t.mux.Lock()
+	defer t.mux.Unlock()
+	t.traceMap[key].LogIfLong(t.Threshold)
+	return
```

Comment:
does the trace need to be deleted at this point from the map so it's garbage collected?

Reply:
We process the same key many times so I didn't like having to delete and recreate.  Recreating keys isn't really a problem so I guess we should garbage collect these...